pkg/handlers/webhookreceiver.go

```diff
@@ -250,18 +253,52 @@ func (h *WebhookReceiverHandler) updateNotificationStatus(n *oav1alpha1.Notifica
 			return err
 		}
 
+		timeNow := &v1.Time{Time: time.Now()}
 		status, err := m.Status.GetNotificationRecord(n.Name)
 		if err != nil {
 			// Status does not exist, create it
+			// This will happen only when the alert is firing first time
 			status = &oav1alpha1.NotificationRecord{
 				Name:                n.Name,
 				ServiceLogSentCount: 0,
 			}
-			status.SetStatus(oav1alpha1.ConditionServiceLogSent, "Service log sent")
+			status.SetStatus(oav1alpha1.ConditionAlertFiring, "Alert is firing", corev1.ConditionTrue, timeNow)
+			status.SetStatus(oav1alpha1.ConditionAlertResolved, "Alert resolved", corev1.ConditionFalse, timeNow)
+			status.SetStatus(oav1alpha1.ConditionServiceLogSent, "Service log sent", corev1.ConditionTrue, timeNow)
 		} else {
 			// Status exists, update it
-			status.SetStatus(oav1alpha1.ConditionServiceLogSent, "Service log sent")
+			// When the alert is already firing
+			firingCondition := status.Conditions.GetCondition(oav1alpha1.ConditionAlertFiring).Status
+			if firingCondition == corev1.ConditionTrue {
+				firedConditionTime := status.Conditions.GetCondition(oav1alpha1.ConditionAlertFiring).LastTransitionTime
+				resolvedConditionTime := status.Conditions.GetCondition(oav1alpha1.ConditionAlertResolved).LastTransitionTime
+				if firing {
+					// Status transition is Firing to Firing
+					// Do not update the condition for AlertFiring and AlertResolved
+					// Only update the timestamp for the ServiceLogSent
+					status.SetStatus(oav1alpha1.ConditionAlertFiring, "Alert is firing", corev1.ConditionTrue, firedConditionTime)
+					status.SetStatus(oav1alpha1.ConditionAlertResolved, "Alert resolved", corev1.ConditionFalse, resolvedConditionTime)
+					status.SetStatus(oav1alpha1.ConditionServiceLogSent, "Service log sent", corev1.ConditionTrue, timeNow)
+				} else {
+					// Status transition is Firing to Resolved
+					// Update the condition status and timestamp for AlertFiring
+					// Update the condition status and timestamp for AlertResolved
+					// Update the timestamp for the ServiceLogSent
+					status.SetStatus(oav1alpha1.ConditionAlertFiring, "Alert is firing", corev1.ConditionFalse, timeNow)
+					status.SetStatus(oav1alpha1.ConditionAlertResolved, "Alert resolved", corev1.ConditionTrue, timeNow)
+					status.SetStatus(oav1alpha1.ConditionServiceLogSent, "Service log sent", corev1.ConditionTrue, timeNow)
+				}
```

Comment:
So from the flow I see that `processAMReceiver` method would called `processAlert` method in a loop for only the firing alerts. This is why, the newly added bool value for `updateNotificationStatus` is set to `true` always. Considering that, will this highlighted code in the `else` block be ever reached?

Reply:
My guess is this is being written in advance of OSD-10225 to implement `resolved` handling (which would make the same call, but with `false`), but Meng Bo will be in the best position to advise.

Reply:
Yes, the block for the `!firing` is used for the resolve part in OSD-10225.
I am just thinking consider them together will be easier from the logic over all.

And when we are going to implement the resolve alert logic.
We can just call the same function `updateNotificationStatus()` with a `false` in the end, just like Matt mentioned.

We may need to do some small tweak in `processAlert()` function then if we want to use it in the loop `for _, alert := range d.Alerts.Resolved()` like what we are doing now in L107 for firing alerts.